packages/sanity/src/core/studio/components/navbar/workspace/WorkspaceAuth/WorkspaceAuth.tsx

```diff
@@ -17,10 +17,13 @@ export function WorkspaceAuth() {
   const [selectedWorkspaceName, setSelectedWorkspaceName] = useState<string | null>(
     activeWorkspace?.name || null
   )
-  const selectedWorkspace =
-    workspaces.length === 1
-      ? workspaces[0]
-      : workspaces.find((workspace) => workspace.name === selectedWorkspaceName)
+  const selectedWorkspace = useMemo(
+    () =>
+      workspaces.length === 1
+        ? workspaces[0]
+        : workspaces.find((workspace) => workspace.name === selectedWorkspaceName),
+    [workspaces, selectedWorkspaceName]
+  )
```

Comment:
Workspaces will usually contain < 5 items - is looping over 5 items going to be slower than the overhead of a memo hook here?